[typo](docs) Update docker-doris in developer

Yankee24 opened the pull request:
## Problem Summary:

Update outdated sections in doris-docker.md

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

Yankee24 committed:
update docker-dev

Yankee24 committed:
update docker-dev

jackwener reviewed (dismissed):
LGTM
Thanks!

jackwener reviewed (commented):
Please fix the conflict and I will promote this PR to merge.

chenlinzhong committed:
[remote-udaf](sample) add some c++ demo (#11761)

xinyiZzz committed:
[enhancement](memory) Fix too much cache leads to less memory available for queries (#11751)

Disable Chunk Allocator in Vectorized Allocator, this will reduce memory cache.

For high concurrent queries, using Chunk Allocator with vectorized Allocator can reduce the impact of gperftools tcmalloc central lock.

Jemalloc or google tcmalloc have core cache, Chunk Allocator may no longer be needed after replacing gperftools tcmalloc.

ByteYue committed:
[Fix](audit) do not use duplicate query id in fe.audit.log (#11746)

The original logic in ConnectProcessor.java might result in duplicate query id for different query statement in fe.audit.log as follows.

chenlinzhong committed:
[remote-udaf](sample) add some java demo (#11752)

platoneko committed:
[enhancement](status) Fix Status related macros to enable RVO or move ctor (#11753)

cambyzju committed:
[Enhancement](array-type) enable_array_type flag update (#11785)


Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

eldenmoon committed:
[Enhancement](stream-load-json) using simdjson to parse json (#11665)

Currently we use rapidjson to parse json document, It's fast but not fast enough compare to simdjson.And I found that the simdjson has a parsing front-end called simdjson::ondemand which will parse json when accessing fields and could strip the field token from the original document, using this feature we could reduce the cost of string copy(eg. we convert everthing to a string literal in _write_data_to_column by sprintf, I saw a hotspot from the flamegrame in this function, using simdjson::to_json_string will strip the token(a string piece) which is std::string_view and this is exactly we need).And second in _set_column_value we could iterate through the json document by for (auto field: object_val) {xxx}, this is much faster than looking up a field by it's field name like objectValue.FindMember("k1").The third optimization is the at_pointer interface simdjson provided, this could directly get the json field from original document.

wsjz committed:
[feature-wip](parquet-reader) update column read model and add page index (#11601)

platoneko committed:
Fix nullptr in perform_remote_tablet_gc (#11820)

cambyzju committed:
[fix](str_to_date) str_to_date support format without leading zero (#11817)

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

sohardforaname committed:
[Feature](nereids)support view and nested view (#11589)

support view in query
and add a rewrite rule: merge consecutive projects.
the rule can merge relative consecutive projects to one project to improve efficiency

Lchangliang committed:
[Bugfix](light-shema-change) fix _finish_clone dead lock (#11823)

In engine_clone_task.cpp, it use tablet->tablet_schema() to create rowset, but in the method, it need a lock that already locked in engine_clone_task.cpp:514. It use cloned_tablet_meta->tablet_schema() originally, but modified in #11131. It need to revert to use cloned_tablet_meta->tablet_schema().

yiguolei committed:
[refactor](dpp) remove original dpp writer (#11838)

Co-authored-by: yiguolei <yiguolei@gmail.com>

wangbo committed:
[performance] (vectorization)optimize In Expr (#11826)



Co-authored-by: Wang Bo <wangbo36@meituan.com>

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) fix that sort segments by segment id in descending order (#11811)

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) unique key table with MOW supports sequence column (#11808)

Gabriel39 committed:
[feature](compaction) allow to set disable_auto_compaction for tables (#11743)

JNSimba committed:
[doc](flink-connector) update flink connector 1.15 support (#11824)

update flink connector 1.15 support

yuanyuan8983 committed:
[docs] Change JDBC error port (#11809)

 Change JDBC error port

LemonLiTree committed:
fix-doc (#11756)

Document typo update

Gabriel39 committed:
[Bug](compile) fix compiling problem (#11851)

fix compiling problem

zhangstar333 committed:
[vectorized](improvement) improve agg function of bitmap_union with f… (#11822)

* [vectorized](improvement) improve agg function of bitmap_union with fastuinon

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) unique key table with MOW supports delete sign column (#11672)

Jibing-Li committed:
[Bug](external iceberg table)Fix iceberg on ha-hdfs unknown hostname bug. #11844

yangzhg committed:
[docs](typo) fix some typo in bitmap docs (#11850)

fix some typo in bitmap docs

morrySnow committed:
fix reorder error (#11854)

join reorder throw unexpected exception when join type is not cross and inner

ByteYue committed:
[enhancement](RowDescriptor) enhance tuple_idx check during runtime (#11835)

arcyda committed:
[doc] Fix typos (#11852)

fix a typo in get-starting doc

yixiutt committed:
[feature-wip](unique-key-merge-on-write) add min/max key in segment (#11830)

some feature:
1. add min max key in segment footer to speed up get_row_ranges_by_keys
2. do not load pk bloom filter in query

Co-authored-by: yixiutt <yixiu@selectdb.com>

wangshuo128 committed:
[Enhancement](Nereids) Refine nereids parser. (#11839)

1. Use ParseException in nereids parser.
2. Add check utils in the parser test.
3. Distinguish matchesFromRoot and matches when checking plans.

dataroaring committed:
[improvement](test) add dryRun option and group all cases into either p0 or p1 (#11576)

1. add dryRun option to list tests
2. group all cases into p0 p1 p2

HappenLee committed:
[Feature] (ODBC) support read/write emoji of utf16 via odbc table (#11863)


Co-authored-by: lihaopeng <lihaopeng@baidu.com>

ByteYue committed:
[enhancement](VSlotRef) enhance column_id check in execute function during runtime (#11862)

The column id check in VSlotRef::execute function before is too strict for fuzzy test to continuously produce random query. Temporarily loosen the check logic.
Moreover, there exists some careless call to VExpr::get_const_col, it might return a nullptr but not every function call checks if it's valid. It's an underlying problem.

zxealous committed:
[fix]Fix the coredump when an IOError occurs in be (#11857)

starocean999 committed:
[fix](orderby)remove useless null literal in order by (#11821)

eldenmoon committed:
(vec-stream-load-json) simdjson throw execption lead to core dump (#11880)

when config::enable_simdjson_parser=true in vec streamload, may lead to core dump when json input invalid format string like '{ "a', or all the fields is null like '{}', this may lead to simdjson lib throw some unhandled expection like `Objects and arrays can only be iterated when they are first encountered`.We should take care of these cases

Signed-off-by: eldenmoon <15605149486@163.com>

stalary committed:
[Enhancement](Doe) Be query es use fe generate dsl.  (#11840)

hf200012 committed:
[typo](doc)Add the actual hive bitmap udf documentation (#11883)

add miss hive bitmap udf

wangshuo128 committed:
[Enhancement](Planner) Improve error message when columns order and keys orders don't match. (#11724)

When creating table like this:
```
CREATE TABLE `test`.`test_key_order` (
  `k1` tinyint(4) NULL COMMENT "",
  `k2` smallint(6) NULL COMMENT "",
  `k3` int(11) NULL COMMENT "",
  `v1` double MAX NULL COMMENT "",
  `v2` float SUM NULL COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`k1`, `k3`, `k2`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k1`) BUCKETS 5
PROPERTIES (
"replication_num" = "1"
);
```

The error message before is:
```
Key columns should be a ordered prefix of the schema.
```

With this PR, the error message is:
```
Key columns should be a ordered prefix of the schema. KeyColumns[1] (starts from zero) is k3, but corresponding column is k2 in the previous columns declaration.
```

yinzhijian committed:
[typo](fix) Fix spelling errors in comments (#11810)

HappenLee committed:
[Vectorized](compaction) filter delete data in base compaction (#11721)

* [Vectorized](compaction) filter delete data in base compaction


Co-authored-by: lihaopeng <lihaopeng@baidu.com>

nextdreamblue committed:
[improvement](meta) sort result by tablename when show table status like 'show data' (#11885)

xinyiZzz committed:
[enhancement](memtracker) Optimize readability of mem exceed limit error message #11877

carlvinhust2012 committed:
[fix](array-type) support to insert the largeint in array (#11868)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

BiteTheDDDDt committed:
[Bug](aggregation) fix core dump on 2nd phase aggregate (#11843)

Gabriel39 committed:
[Improvement](load) turn `enable_vectorized_load` on by default (#11833)

Gabriel39 committed:
[Bug](runtime filter) Fix bug for runtime filter in concurrent scanners (#11848)

starocean999 committed:
[fix](orderby)disallow hll and bitmap data type in order by list (#11837)

Kikyou1997 committed:
[Planner](fix)Fix unexpected index out of bound exception (#11819)

BiteTheDDDDt committed:
[Bug](Vectorzed alter table)modify schema change cast validate (#11864)

geniusjoe committed:
[Enhancement](docs) fix document typo #11818

Co-authored-by: ninjazhou <ninjazhou@tencent.com>

adonis0147 committed:
[chore](workflow) Add shellcheck to check shell scripts (#11744)

wsjz committed:
[feature-wip](parquet-reader) row group reader ut finish (#11887)

Co-authored-by: jinzhe <jinzhe@selectdb.com>

jackwener committed:
[fix](Nereids): fix and enable stats derive job (#11755)

fix and enable statistics derive job
Add mock for statistics in computeScan().

a19920714liou committed:
[doc](fix) add STRING conversion notice (#11895)

dataroaring committed:
[impovement](sink) print load_id when sink fails (#11893)

yinzhijian committed:
[fix](function) Fix semantic analysis error in window function at first_value (#11855)

xinyiZzz committed:
[fix](memtracker) Fix PartitionedAggregationNode DCHECK when mem exceed limit (#11902)

chenlinzhong committed:
[remote-udaf](optimize) Optimize RPC exception handling logic (#11680)

BiteTheDDDDt committed:
[Feature](vectorized alter table) set vectorized alter table to default open (#11897)

Gabriel39 committed:
[Regression](datev2) Add test cases for datev2/datetimev2 (#11831)

yixiutt committed:
[bugifx](compaction) fix filter_delete if schema has sequence column (#11909)

introduced in #11721. Use last column as delete sign, but if sequence column
exist, it's wrong.

Co-authored-by: yixiutt <yixiu@selectdb.com>

yixiutt committed:
[Enhancement](load) add hidden_columns in stream load param (#11625)

Stream load will ignore invisible columns if no http header columns
specified, but in some case user cannot get all columns if columns
changed frequently。
Add a hidden_columns header to support hidden columns import。User can
set hidden_columns such as __DORIS_DELETE_SIGN__ and add this column
in stream load data so we can delete this line.
For example:
curl -u root -v --location-trusted -H "hidden_columns: __DORIS_DELETE_SIGN__" -H
"format: json" -H "strip_outer_array: true" -H "jsonpaths: [\"$.id\",
\"$.name\",\"$.__DORIS_DELETE_SIGN__\"]" -T 1.json
http://{beip}:{be_port}/api/test/test1/_stream_load

Co-authored-by: yixiutt <yixiu@selectdb.com>

yew1eb committed:
[Chore](benchmark) Fix benchmark scripts, cover case that $PASSWORD not empty (#11486)

Fix benchmark scripts, cover case that $PASSWORD not empty

luzhijing committed:
[docs](fix) updata pictures licks (#11890)

updata pictures licks

jackwener committed:
[refactor](planner): improve enfocer job. (#11922)

- handle enforce distribution when meet sort.
- calculate stats in enforcer job
- refactor calculate stats.

chenlinzhong committed:
[docs](update)udaf now supported (#11929)

BiteTheDDDDt committed:
[doc](developer-guide) add vscode-lldb usage to developer guide (#11923)

* add codelldb usage

* update

Jibing-Li committed:
[improvement](dependency) Use release mode to build cctz. Improve performance.

adonis0147 committed:
[fix](script) Support starting BE without Java environment (#11910)

carlvinhust2012 committed:
[optimization](array-type) support the decimal/datetime as the nest type of array in print_value (#11784)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

Gabriel39 committed:
[Bug](shell) fix wrong condition expression in script (#11913)

* [Bug](shell) fix wrong condition expression in script

BiteTheDDDDt committed:
[Bug](function) fix dcheck fail on close vexpr ctx (#11908)

jackwener committed:
[refactor](planner): refactor and replace use NIO (#11645)

* [refactor](planner): refactor equals code in Catalog dir.

yangzhg committed:
[Bugfix](datetime) fix DateLiteral range check is no longer valid (#11917)

* [Bugfix](datetime) fix DateLiteral range check is no longer valid

carlvinhust2012 committed:
[fix](array-type) run 'show create table' return null (#11912)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

adonis0147 committed:
[fix](workflow) Fix the errors when using sh to run shell scripts (#11898)

Gabriel39 committed:
[regression](datev2) add schema change cases for datev2/datetimev2 (#11924)

GoGoWen committed:
[Bugfix](FE) fix npe issue when exec 'show tablets' #11896

luzhijing committed:
[README](fix) update pictures links in README.md (#11891)

luzhijing committed:
Update basic-summary.md (#11889)

Update basic-summary

adonis0147 committed:
[fix](build) Resolve the conflicts when building be with java-udf (#11938)

Toms1999 committed:
[extension](mysql-to-doris) add odbc conf and  some fix (#11692)

jackwener committed:
[refactor](planner): refactor equals code in Catalog dir. (#11903)

yangzhg committed:
[bugfix](load) fix cancel load stmt cannot recognize key words in upper case (#11906)

yangzhg committed:
[Bugfix](inpredicate) fix in predicate in group by clause may cause NPE (#11886)

* [bug](inpredicate) fix in predicate in group by clause may cause NPE

BiteTheDDDDt committed:
[Bug](cast) change binary predicate finally cast to varchar (#11796)

jackwener committed:
[improve](fe): use `Pair.of` to replace `new Pair<>()` (#11945)

jackwener committed:
[improve](nereids): remove FakeJoin.java (#11946)

xinyiZzz committed:
[fix](memtracker) Remove useless memory exceed check #11939

cambyzju committed:
[refractor](bitmap) bitmap serialize and deserialize refractor (#11921)


Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

xinyiZzz committed:
[fix](memtracker) Improve memory tracking accuracy for exec nodes (#11947)

hf200012 committed:
[feature](spark-load)Spark load supports string type data import (#11927)

mrhhsg committed:
[improvement](agg) Serialize the fixed-length aggregation results with corresponding columns instead of ColumnString (#11801)

AshinGau committed:
[feature-wip](parquet-reader) decode parquet time & datetime & decimal (#11845)

1. Spark can set the timestamp precision by the following configuration:
spark.sql.parquet.outputTimestampType = INT96(NANOS), TIMESTAMP_MICROS, TIMESTAMP_MILLIS
DATETIME V1 only keeps the second precision, DATETIME V2 keeps the microsecond precision.
2. If using DECIMAL V2, the BE saves the value as decimal128, and keeps the precision of decimal as (precision=27, scale=9). DECIMAL V3 can maintain the right precision of decimal

hf200012 committed:
[improvement](config)Enable insert strict (#11866)

xinyiZzz committed:
[enhancement](memory) Reduce virtual memory used by PaddedPODArray (#11816)

amosbird committed:
[fix][Vectorized] Fix nullptr deref in data sink (#11473)

brpc cache may return nullptr.

starocean999 committed:
[fix](agg)repeat node shouldn't change slot's nullable property of agg node (#11859)

mrhhsg committed:
[improvement](compress) Support compress/decompress block with lz4 (#11955)

Gabriel39 committed:
[regression](rollup) Modify test case (#11960)

dataroaring committed:
[FixAssist](compaction) add DCHECK in BlockReader::_unique_key_next_block to reason problem (#11951)

morrySnow committed:
[feature](Nereids)add type coercion rule for nereids (#11802)

- add an interface ExpectsInputTypes to Expression
- add an interface ImplicitCastInputTypes to Expression
- add a Expression rewrite rule for type coercion
- add a Check Analysis Rule to check whether Plan is Semantically correct

if Expression implements ImplicitCastInputTypes, type coercion rule will automatic rewrite its children that casting it to the most suitable type.
If Expression implements ExpectsInputTypes, Check Analysis will check its children's type whether accepted by expects input types.

jackwener committed:
[feature](nereids): enable exploration job (#11867)

Enable the exploration job, and fix related problem.

correct the join reorder

pingchunzhang committed:
[github](checks) change the requirement of github checks (#11978)

morningman committed:
[feature-wip](new-scan) add scanner scheduling framework (#11582)

There are currently many types of ScanNodes in Doris. And most of the logic of these ScanNodes is the same, including:

Runtime filter
Predicate pushdown
Scanner generation and scheduling
So I intend to unify the common logic of all ScanNodes.
Different data sources only need to implement different Scanners for data access.
So that the future optimization for scan can be applied to the scan of all data sources,
while also reducing the code duplication.

This PR mainly adds 4 new class:

VScanner
All Scanners' parent class. The subclasses can inherit this class to implement specific data access methods.

VScanNode
The unified ScanNode, and is responsible for common logic including RuntimeFilter, predicate pushdown, Scanner generation and scheduling.

ScannerContext
ScannerContext is responsible for recording the execution status
of a group of Scanners corresponding to a ScanNode.
Including how many scanners are being scheduled, and maintaining
a producer-consumer blocks queue between scanners and scan nodes.

ScannerContext is also the scheduling unit of ScannerScheduler.
ScannerScheduler schedules a ScannerContext at a time,
and submits the Scanners to the scanner thread pool for data scanning.

ScannerScheduler
Unified responsible for all Scanner scheduling tasks

Test:
This work is still in progress and default is disabled.
I tested it with jmeter with 50 concurrency, but currently the scanner is just return without data.
The QPS can reach about 9000.
I can't compare it to origin implement because no data is read for now. I will test it when new olap scanner is ready.
Co-authored-by: morningman <morningman@apache.org>

catpineapple committed:
[fix](error-code) prompt error when MySQL client login password is incorrect #11973

Henry2SS committed:
[Bug](error code) fix db access error code msg #11962

Co-authored-by: wuhangze <wuhangze@jd.com>

yixiutt committed:
[feature-wip](unique-key-merge-on-write) opt lock and only save valid delete_bitmap (#11953)

1. use rlock in most logic instead of wrlock
2. filter stale rowset's delete bitmap in save meta
3. add a delete_bitmap lock to handle compaction and publish_txn confict

Co-authored-by: yixiutt <yixiu@selectdb.com>

jacktengg committed:
[bugfix](dict) fix coredump of dict colum range predicate when there is null value (#11967)

jackwener committed:
[refactor](Nereids): merge rewrite job (#11950)

merge rewrite job

HappenLee committed:
[bug](compaction) fix bug of coredump of filter delete chose wrong filter column (#12002)

* [bug](compaction) fix bug of coredump of filter delete chose wrong filter column

* clang format

carlvinhust2012 committed:
[fix](array-type) fix be occasional coredump when use stream load (#11997)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

morningman committed:
[minor](runtime-filter) add DCHECK for runtimefilter bug (#11996)

Not a fix, just add debug info to try find root cause of #11995

dataroaring committed:
[improvement](start_script) add ASAN and UBSAN env in start_be.sh #12014

Neither asan nor ubsan does not generate core file by default,
however, we need core file to analyze problems detected by asan and ubsan.

stalary committed:
[Bug](ctas/ctl): Ctas and ctl default value error (#12015)

AshinGau committed:
[fix](auth) no need to degrade USAGE_PRIV in userPrivTable (#12016)

morrySnow committed:
[chore](checkstyle)forbidden import all kind of relocated guava (#12018)

starocean999 committed:
[fix](join)the hash join node would get the wrong nullable if the child node is cross join node (#11971)

* [fix](join)the hash join node would get the wrong nullable if the child node is cross join node

bin41215 committed:
[typo](doc) fix some typo in faq docs #11977

luozenglin committed:
[fix](having) Fix the `with as` clause containing having caused a null pointer (#12007)

xinyiZzz committed:
[enhancement](memory) Remove unused reservation tracker (#11969)

hf200012 committed:
[deps](httpv2)upgrade springboot version to 2.7.3 (#11963)

starocean999 committed:
[fix](agg)grouping_id function in having is not correctly substituted (#11834)

carlvinhust2012 committed:
[fix](array-type) fix the be core dump when import number larger than uint64 (#11853)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

HappenLee committed:
[Bug](bitmap) intersect_count function use in string cause ASAN error (#11936)

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) fix that IndexedColumnIterator next batch may return empty result (#11928)

morningman committed:
[doc](faq) add Problem with the SSL CA cert FAQ (#11982)

xiaojunjie committed:
[enhancement](Nereids)support max function (#11795)

- add aggregate function max to Nereids
- add function P0 regression test for Nereids

hf200012 committed:
[typo](doc)Adding Missing Documents (#12022)

Adding Missing Documents

Userwhite committed:
[Improvement](storage) reuse schema and rowblockv2 on single scanner_thread (#11392)

* support reuse rowblockv2 on single thread

xiaojunjie committed:
[regression-test](p0) add case for window function at min/max when vectorized (#11989)

Co-authored-by: xiaojunjie <xiaojunjie@baidu.com>

xinyiZzz committed:
[fix](memtracker) Improve performance of tracking real physical memory of PodArray #12021

starocean999 committed:
[fix](sort)the sort expr nullable info is wrong in some case (#12003)

hello-stephen committed:
[tools](ssb and tpch)optimize tools (#11975)

BiteTheDDDDt committed:
[Chore](script) support custom python version on build script (#11954)

mrhhsg committed:
[fix](agg) Crashing caused by serialization in streaming aggregation (#12027)

starocean999 committed:
[fix](grouping)fix grouping function bug (#11861)

HappenLee committed:
[bug](vectorized) fix bug of tuple is null null side do not set (#12012)

luozenglin committed:
[enhancement](ldap) optimize LDAP authentication. (#11948)

* [enhancement](ldap) optimize LDAP authentication.

1. Support caching LDAP user information.
2. HTTP authentication supports LDAP.
3. LDAP temporary users support default user property.
4. LDAP configuration supports the `admin show config` and `admin set config` commands.

caiconghui committed:
[fix](ut) Fix fe ut npe for DeleteHandlerTest (#12033)

zxealous committed:
[Fix](remote) Fix thread safety issue in cache (#11984)

morrySnow committed:
[feature](Nereids)add normalize aggregate rule (#12013)

yuanyuan8983 committed:
[typo](doc) Change wrong words #11778

starocean999 committed:
[fix](union)the result exprs of union node should substitute by child node's smap (#11933)

union node's result exprs should be substitued by child node's smap first, then the following "computePassthrough" method would have correct information to do its job.

yixiutt committed:
[feature-wip](unique-key-merge-on-write) add regression case (#11731)

Co-authored-by: yixiutt <yixiu@selectdb.com>

xy720 committed:
[Fix] fix `cast(array<not_null> as array<>)` causes be core dump (#11649)

stalary committed:
[Bug](ctas): Ctas use default DEFAULT CURRENT_TIMESTAMP insert error (#12056)

Gabriel39 committed:
[minor](conjuncts) remove row-based conjuncts from vectorized engine (#12053)

englefly committed:
[feature](Nereids) support non-equal predicates in Join (#11812)

Currently, the join on conditions are denoted by Join.conditions as one expression.
It is not convenient to get equal-predicates for hash table.

This pr add a rule FindHashConditionForJoin to split on-conditions into two parts:
- hashJoinPredicates
- otherJoinConditions
And store them as two attributes in LogicalJoin and PhysicalJoin.

JoinReorder rule is impacted by this change, since the way we get the equal conditions between join children are changed.
In this pr, we changed the output of multiJoin slightly. Equations like A=1 are moved from join condition to upper filter.
For more detail, refer to fe/fe-core/src/test/java/org/apache/doris/nereids/datasets/ssb/SSBJoinReorderTest.java.

dataroaring committed:
[improvement](doc test) add docs for window_funnel and refine tests (#12059)

add docs for window_funnel and refine tests

BiteTheDDDDt committed:
[Enchancement](optimize) set result_size_hint to filter_block (#11972)

hello-stephen committed:
[ci](shellcheck)shellcheck include ssb-tools and tpch-tools (#12039)

morningman committed:
[fix](scan) use serial scan thread token only for scan node (#12058)

Only the scan node's limit is less than 1024, we can use serial thread token to submit scanners.
Or it will slow down the query.

morningman committed:
[feature](http) get create table stmt for a given query (#11979)

This API can help user to get all create table statement of a given SQL.
So that it will be easier to run the case in other Doris cluster.
See document for details

catpineapple committed:
[feature] (dbt) add incremental and init interactive command line (#11870)

add dbt-doris incremental model and init interactive command line

platoneko committed:
[feature](cold_on_s3) Show remote data usage via SHOW BACKENDS  and SHOW TABLETS statements (#11450)

hf200012 committed:
[typo](doc)Window funnel fix  (#12064)

window funnel fix

morrySnow committed:
[behavior change](planner)change Doris's query organization syntax to standard sql (#9745)

mrhhsg committed:
[improvement](sink) Support local exchange for multi fragment instances (#12017)

morningman committed:
[fix](ut) fix ctas regression test case (#12076)

morningman committed:
[fix](bdbje) add reserved disk config to avoid too many reserved bdbje files (#12043)

Gabriel39 committed:
[Bug](runtime filter) fix bug for late-arrival runtime filters (#12049)

yixiutt committed:
[feature-wip](unique-key-merge-on-write) add pk_uk correctness check case (#12060)

Co-authored-by: yixiutt <yixiu@selectdb.com>

zhannngchen committed:
[feature-wip](unique-key-merge-on-write) support alter table column for MoW (#12052)

hf200012 committed:
[typo](doc)website sidebar Menu order (#12073)

menu order

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) fix that version is awlays 0 when update delete bitmap (#12044)

wsjz committed:
[feature-wip](parquet-reader) parquert scanner can read data (#11970)

Co-authored-by: jinzhe <jinzhe@selectdb.com>

liaoxin01 committed:
[feature-wip](unique-key-merge-on-write) unique key table with MOW supports update (#11882)

zhannngchen committed:
[feature-wip](unique-key-merge-on-write) fix bugs on tablet clone #12067

cambyzju committed:
[Enhancement](table_function) table function node enhancement (#12038)

* table function node enhancement

* also avoid copy for non-vec table function node

* fix table function node output slots calculation while lateral view involves subquery

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

xinyiZzz committed:
[fix](memory) Fix disable_mem_pools to disable cache #12087

bin41215 committed:
[typo](doc) fix some typo in docs #12054

yiguolei committed:
[bugfix](light weight schema change) support delete condition in schema change (#11869)

* [bugfix](light weight schema change) support delete condition in schema change


Co-authored-by: yiguolei <yiguolei@gmail.com>

starocean999 committed:
[fix](projection)join node should always output at least one column (#12080)

xinyiZzz committed:
[fix](memory) fix set disable_chunk_allocator_in_vec=false performance #12092

xinyiZzz committed:
[fix](memtracker) Optimize the return msg of process memory limit exceed #12086

Return the real process memory information when the process exceeds mem limit
Optimize the memory exceed limit log printing logic
process tracker does not participate in process memory limit.

luozenglin committed:
[fix](planner) fix rewrite subquery error (#12075)

BiteTheDDDDt committed:
[Bug](function) fix aggFnParams set not correct (#12006)

englefly committed:
[enhancement](Nereids)merge consecutive LogicalLimit plan (#12096)

This rule aims to merge consecutive limits.

LogicalLimit(limit=10, offset=4)
+---LogicalLimit(limit=3, offset=5)

transformed to

LogicalLimit(limit=3, offset=5)

where

newLimit.limit    = min(topLimit.limit, bottomLimit.limit)
newLimit.offset = bottomLimit.offset
topLimit.offset  is ignored

JNSimba committed:
[sample](flink-connector) add full db sync v2 (#12090)

* add db sync v2

cambyzju committed:
[fix](array-type) add ARRAY_BOOLEAN support for lots of array functions #12079

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

carlvinhust2012 committed:
[fix](array-type) fix the be core dump when use collect_list result to insert (#12045)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

yangzhg committed:
[Bugfix](load) fix be will coredump when parsing malformed json file using simdjson (#12062)

* [Bugfix](load) fix be will coredump when parsing malformed json file using simdjson

jackwener committed:
[fix](Nereids)fix some problem in CBO. (#12083)

This PR fix three problem in Nereids.
- Add selected index, partition and tablet Info in LogicalOlapScan
- with JoinReorderContext in new LogicalJoin
- fix compute data size when no column size info in StatsCaculator

AshinGau committed:
[feature-wip](parquet-reader) parquet dictionary decoder (#11981)

Parse parquet data with dictionary encoding.

Using the PLAIN_DICTIONARY enum value is deprecated in the Parquet 2.0 specification.
Prefer using RLE_DICTIONARY in a data page and PLAIN in a dictionary page for Parquet 2.0+ files.
refer: https://github.com/apache/parquet-format/blob/master/Encodings.md

pengxiangyu committed:
[feature](remote)Add cache files cleaner for remote olap files (#11959)

hf200012 committed:
[typo](doc)Add the backup operation, using the following instructions from S3 (#12091)

 Add the backup operation, using the following instructions from S3

weizuo93 committed:
update document of alter table oartition (#12077)

update document of alter table oartition

zxealous committed:
fix configs default value in docs (#12047)

fix configs default value in docs

smallhibiscus committed:
[regression-test](unique) Add tpch sf0.1 test under unique table with zstd (#12122)

Gabriel39 committed:
[Improvement](load) accelerate streamload and compaction (#12119)

* [Improvement](load) accelerate streamload and compaction

Kikyou1997 committed:
[feature](Nereids)add range partition prune (#11964)

1. Rewrite Filter(Project) to Project(Filter) to make sure when do partition prune the tree looks like this: Project(Filter(OlapScan)).
2. Enable the MergeConsecutiveProject MergeConsecutiveFilter rules.
3. prune range partition just like what Legacy Planner do.

hf200012 committed:
[Typo](doc) Rewrite the Quick Start document (#11956)

AshinGau committed:
[feature-wip](parquet-reader) generate null values and NullMap for parquet column (#12115)

Generate null values and NullMap for the nullable column by analyzing the definition levels.

morningman committed:
[api-change](http) change kill query http api by using query id (#12120)

Now user can cancel query id by http by following steps:

Get query id by trace id
cancel query by query id
The modified api has not been released yet.

carlvinhust2012 committed:
[fix](array-type) fix the wrong data when use stream load to import '\N' (#12102)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

cambyzju committed:
[fix](array-type) adjust enable_array_type config (#12071)

Problem:
1. `enable_array_type` is masterOnly;
2. dynamic open config only affect FE MASTER
`admin set frontend config("enable_array_type"="true");`
3. query in FE FOLLOWER will fail, because of `enable_array_type` is false in FE FOLLOWER
`select * from table_with_array `

Solution:
Only check `enable_array_type` while creating new tables with array column.

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

pengxiangyu committed:
[fix](remote)Fix bug for cache reader (#12104)

jackwener committed:
[test](Nereids): add more plan equals test for Nereids (#12127)

- add more plan equals test for Nereids
- fix join equals bugs

platoneko committed:
[refactor](status) Refactor status handling in agent task (#11940)

Refactor TaggableLogger
Refactor status handling in agent task:
Unify log format in TaskWorkerPool
Pass Status to the top caller, and replace some OLAPInternalError with more detailed error message Status
Premature return with the opposite condition to reduce indention

englefly committed:
[refactor](BE) return error status when vslot_ref contains invalid slot_id (#12106)

In current implementation, we detect invalid slot at execute phase. At execute phase, it is hard to get useful information for further debug. This pr moves error detection ahead to prepare phase, so that we can log related tuple descriptors.

tk047 committed:
[fix](fe) Fixed alterOp from HashSet to EnumSet (#12094)

Change the HashSet to EnumSet of the AlterOp's currentOps for better performance

BiteTheDDDDt committed:
[Bug](lateral-view) fix some conjunct not work on lateral view #12105

yinzhijian committed:
[fix](view) Fix view not showing specific lengths for varchar type (#12107)

Gabriel39 committed:
[Refactor](predicate) Unify Conditions and ColumnPredicate (#11985)

chenlinzhong committed:
[docs](remote-udaf) change title (#12133)

Gabriel39 committed:
[Improvement](window funnel) restrict timestamp to datetime type in window funnel (#12123)

yangzhg committed:
[Bugfix](coredump) fix coredump cause by fmt::format param malformt (#12138)

fix coredump cause by fmt::format param malformt

xy720 committed:
[Docs](array type) Update array type documentation (#12097)

morrySnow committed:
[fix](Nereids)join output order need same with child plan node output when translate (#12130)

In BE, There is an implicit convention that HashJoinNode's left child's output Slot must before right child's output slot in intermediateTuple.

However, after we do commute rule on join plan in Nereids, this convention will be broken and cause core dump in BE.

There are two way to fix this problem:
1. add a project on join after we do commute
2. reorder output of join node when we do translate

Since we cannot translate project yet because BE projection support is on going(#11842). So we use second way to fix it now. After the project translation could work correctly, we should use the first way to fix it.

YangShaw committed:
[feature](nereids)support one expression-rewrite rule: inPredicateToEqualTo (#12046)

Add one expression rewrite rule:
rewrite InPredicate to an EqualTo Expression, if there exists exactly one element in InPredicate Options.

Examples:
1. where A in (x) ==> where A = x
2. where A not in (x) ==> where not A = x

morningman committed:
Revert "[behavior change](planner)change Doris's query organization syntax to standard sql (#9745)" (#12135)

Reverts apache/doris#9745
This may cause NPE when calling `parseDefineExprWithoutAnalyze()`

cambyzju committed:
[fix](array-type) array can not be distributed key and aggregation key (#12082)

Array column should not be distributed key or group by or aggregate key, we should forbid it before create table.
Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

caoliang-web committed:
Failed to get doris_odbc_name value in mysql_to_doris, replace driver value error (#11965)

Failed to get doris_odbc_name value in mysql_to_doris, replace driver value error

xinyiZzz committed:
[fix](memtracker) Fix mem limit exceed return wrong format (#12139)

jackwener committed:
[improve](Nereids)make some DataType's constructor's access level to private (#12143)

include these types:
- NullType
- BooleanType
- TinyIntType
- SmallIntType
- IntegerType
- BigIntType
- LargeIntType
- FloatType
- DoubleType
- DateType
- DateTimeType
- StringType

caiconghui committed:
[Enhancement](select) Make select variables request handled in fe without be to avoid potential blocked problem when login (#12111)

adonis0147 committed:
[chore](git) Some paths should be tracked by git (#12072)

FreeOnePlus committed:
Fixed incorrect description of column length limit in partition bucketing documentation. (#12154)

Signed-off-by: manyi <fop@freeoneplus.com>

Signed-off-by: manyi <fop@freeoneplus.com>
Co-authored-by: manyi <fop@freeoneplus.com>

BiteTheDDDDt committed:
[Enhancement](compaction) add compaction use time count (#12141)

yiguolei committed:
[bugfix](delete hanlder) delete predicate is merged and could not find schema cause core dump (#12161)

Co-authored-by: yiguolei <yiguolei@gmail.com>

xinyiZzz committed:
[enhancement](memtracker) Improve performance of tracking real physical memory of PODArray #12168

morningman committed:
[feature-wip](scan) add profile for new olap scan node (#12042)

Copy most of profiles from VOlapScanNode and VOlapScanner to NewOlapScanNode and NewOlapScanner.
Fix some blocking bug of new scan framework.
TODO:

Memtracker
Opentelemetry spen
The new framework is still disabled by default, so it will not effect other feature.

JNSimba committed:
[Doc](flink-connector) update sample code url (#12149)

update sample code url

hf200012 committed:
[typo](doc)Modify data operation documentation (#12137)

fix data operate doc

chenlinzhong committed:
[docs](hive)add hadoop.username config item (#12109)

add hadoop.username config item

dataroaring committed:
[fix](planner) let OlapScanNode turn off preaggragation when there is a filter on DELETE_SIGN (#12118)

We can skip aggregate on replace column, otherwise it would generate
wrong result. e.g. a row in UNIQUE is deleted by delte_sign_column,
then it would be returned.

Kikyou1997 committed:
[feature](Nereids)add the ability of projection on each ExecNode and add column prune on OlapScan (#11842)

We have added logical project before, but to actually finish the prune to reduce the data IO, we need to add related supports in translator and BE.
This PR:
- add projections on each ExecNode in BE
- translate PhysicalProject into projections on PlanNode in FE
- do column prune on ScanNode in FE

Co-authored-by: HappenLee <happenlee@hotmail.com>

xy720 committed:
[Bug](compile) fix clang build for be (#12183)

iceqing committed:
[docs](query plan) Add a description `explain graph` and `desc graph` and `explain verbose` (#11697)

* docs: 增加explain graph与desc graph描述，避免文档前后描述不一致

* docs: add the usage docs of `explain verbose`

catpineapple committed:
[fix] (docs) Fix Data type: string length and parameter description (#11565)

924060929 committed:
[refactor](Nereids)refactor memo.copyIn (#12147)

this pr do 2 refactor
1. remove useless parameter from `Plan#computeOutput`
2. refactor memo.copyIn

It the past, `memo.copyIn` has complex logic to process init, rewrite and copyIn, It's difficult to understand and easy to meet bug and leak memory for some unreachable group/groupExpression.

So I separate it into three methods:
1. `Memo.init` for init Memo by LogicalPlan
2. `Memo.doRewrite` for rewrite
3. `Memo.doCopyIn` for exploration and implementation

And separate the UT into 3 files
1. `MemoInitTest`
2. `MemoRewriteTest`
3. `MemoCopyInTest`

I have added a lots of UT for `Memo.rewrite`, and add some unreachable DAG check in the PlanChecker, when the plan is changed.

morrySnow committed:
[fix](Nereids)Sum return DoubleType when child is DecimalType by mistake (#12169)

When Sum's child is Decimal, Return Double Type by mistake lead to result error, so we should keep the return type to decimal when the child expression's type is decimal.

yangzhg committed:
[fix](simdjson) disable simdjson by default, wait for fix the bug (#12179)

zhangshuocn committed:
fix typos in get-starting.md (both en and zh-CN) (#12189)

jackwener committed:
[feature](Nereids)enable CBO optimize stage in Nereids (#12008)

- enable CBO stage in Nereids
- use the `chooseBestPlan()` to get the best plan
- add a new rule JoinCommuteProject
- test the stage by JoinCommute rule

morrySnow committed:
[fix](Nereids)get NPE from finalize TimestampArithmeticExpr that generated by ExpressionTranslator (#12163)

This PR:
1. refactor getDataType in TimestampArithmetic
2. set TimeUnit correctly when translate TimestampArithmetic to TimestampArithmeticExpr

morningman committed:
[feature](stmt) add ADMIN COPY TABLET stmt for local debug (#12176)

Add a new stmt ADMIN COPY TABLET for easy copy a tablet to local env to reproduce problem.
See document for more details.

englefly committed:
[fix](planner) a slot id is bounded on a wrong tuple id, if cross join has a hash join as child (#12156)

xinyiZzz committed:
[fix](compile) Fix compile error '<unknown>' may be used uninitialized in PODArray::insert_prepare #12202

morningman committed:
[refactor](column) remove confused field (#12187)

hou-rong committed:
[performance improvement] Spark Load, SparkDpp processRDDAggregate performance improvement (#12186)

Co-authored-by: hourong <hourong@zhihu.com>

zxealous committed:
[optimize](remote) Optimize cache reader use a pre-created buffer when downloading the cache (#12165)

* optimize cache reader

* add description for config

* optimize cache reader

* optimize cache reader

cambyzju committed:
[enhancement](array-type) Array type do not support compare with '=','>', '<', make the error message more readable (#12181)

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

jackwener commented:
please use git rebase.
Now the git history is mess up.

Yankee24 committed:
Modify Docker documents.

jackwener commented:
`Files changed 4,972`

Yankee24 committed:
Modify Docker documents.

Yankee24 committed:
Modify Docker documents.

jackwener commented:
Look like you don't be familiar with git.
I think the easiest way is create a new branch from upstream master and copy the modification into it.
